src/sqlfluff/core/rules/base.py

```diff
@@ -802,6 +802,11 @@ def _choose_anchor_segment(
 
         anchor: BaseSegment = segment
         child: BaseSegment = segment
+        if (
+            context.parent_stack[0]
+            and context.parent_stack[0].path_to(segment)
+            and context.parent_stack[0].path_to(segment)[1:-1]
+        ):
```

Comment:
Should the lines below this be indented? This looks like a syntax error (an `if` statement with no body).